Lombiq.Hosting.Tenants.Admin.Login/Services/TenantLoginKeyValidator.cs

```diff
@@ -16,16 +16,14 @@ private static string GeneratePassword(int length)
         {
             const string validCharacters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890!#";
             var stringBuilder = new StringBuilder();
-            using (var random = new RNGCryptoServiceProvider())
-            {
-                byte[] uintBuffer = new byte[sizeof(uint)];
+            using var random = RandomNumberGenerator.Create();
+            byte[] uintBuffer = new byte[sizeof(uint)];
 
-                while (length-- > 0)
-                {
-                    random.GetBytes(uintBuffer);
-                    uint randomNumber = BitConverter.ToUInt32(uintBuffer, 0);
-                    stringBuilder.Append(validCharacters[(int)(randomNumber % (uint)validCharacters.Length)]);
-                }
+            while (length-- > 0)
+            {
+                random.GetBytes(uintBuffer);
```

Comment:
The broken arrow in the `while` makes me feel uncomfortable. Could you move the decrementation into a separate line?
```suggestion
            while (length > 0)
            {
                length--;
                random.GetBytes(uintBuffer);
```

Reply:
Since in military lingo broken arrow means "an accident that involves nuclear weapons or nuclear weapons components, but does not create the risk of nuclear war" I think you being uncomfortable is justified.